webapp/src/app.tsx

```diff
@@ -1857,14 +1862,19 @@ export class ProjectView
             opts.trace = true;
 
         this.stopSimulator();
+        if (opts.cancellationToken && opts.cancellationToken.isCancelled()) return Promise.resolve();
 
         const state = this.editor.snapshotState()
         return compiler.compileAsync(opts)
             .then(resp => {
                 this.clearSerial();
                 this.editor.setDiagnostics(this.editorFile, state)
+                if (opts.cancellationToken && opts.cancellationToken.isCancelled()) return;
                 if (resp.outfiles[pxtc.BINARY_JS]) {
-                    simulator.run(pkg.mainPkg, opts.debug, resp, this.state.mute, this.state.highContrast, pxt.options.light, clickTrigger)
+                    simulator.run(pkg.mainPkg, opts.debug, resp, this.state.mute, this.state.highContrast, pxt.options.light, opts.clickTrigger)
+                    if (opts.cancellationToken && opts.cancellationToken.isCancelled()) {
+                        this.stopSimulator();
+                    }
                     this.setState({ running: true, showParts: simulator.driver.runOptions.parts.length > 0 })
```

Comment:
Setting state running after stopping simulator

Reply:
Yeah I'm seeing that too.